src/diracx/client/aio/_patch.py

```diff
@@ -6,11 +6,30 @@
 
 Follow our quickstart for examples: https://aka.ms/azsdk/python/dpcodegen/python/customize
 """
-from typing import List
+import asyncio
+import json
+from types import TracebackType
+import requests
+from datetime import datetime, timedelta, timezone
+from pathlib import Path
+from typing import Any, List, Optional
+from azure.core.credentials import AccessToken
+from azure.core.credentials_async import AsyncTokenCredential
+from azure.core.pipeline import PipelineRequest
+from azure.core.pipeline.policies import AsyncBearerTokenCredentialPolicy
 
-__all__: List[
-    str
-] = []  # Add all objects you want publicly available to users at this package level
+from diracx.client.models import DeviceFlowErrorResponse, TokenResponse
+from diracx.core.preferences import get_diracx_preferences
+
+from ._client import Dirac as DiracGenerated
+
+__all__: List[str] = [
+    "DiracClient",
+]  # Add all objects you want publicly available to users at this package level
+
+
+CREDENTIALS_PATH = Path.home() / ".cache" / "diracx" / "credentials.json"
```

Comment:
These settings probably should be in one of:
* `diracx.cli.__init__.py` because the CLI is the primary user for that location
*  `diracx.client.__init__.py` because so anybody using the client (cli, legacy dirac, tasks) may potentially use it
* `diracx.core.__init__.py`  because it it really is such a crucial value that it could be here

Reply:
I vote for the last one (and maybe naming it `DEFAULT_CREDENTIALS_PATH`)

Reply:
I actually had a chat with @aldbr and actually what could make even more sense is to put it in the `Preferences`.

Reply:
I put it in the `Preferences`